#!/usr/bin/env python3

# # I have neither given nor received any unauthorizedaid in completing this work,
# # nor have I presented someone else's work as my own.
# # Your Name: Devson Butani
# # LTU ID: 000732711
# # Date: 12/13/2022

PACKAGE = "term_project_pkg"  # Catkin Package Name

# Import library
from dynamic_reconfigure.parameter_generator_catkin import *

# Create Instance
gen = ParameterGenerator()

gen.add(
    "edge_detect_blue_percent",  # variable name
    double_t,  # data type
    0,  # msg level
    ">>> Minimum required BLUE % (edge color) for edge detected status",  # display on hover
    2,  # default
    0,  # min
    100,  # max
)

gen.add(
    "mat_detect_white_percent",  # variable name
    double_t,  # data type
    0,  # msg level
    ">>> Minimum required WHITE % (out of mat status)",  # display on hover
    65,  # default
    0,  # min
    100,  # max
)

gen.add(
    "plow_capacity",  # variable name
    double_t,  # data type
    0,  # msg level
    ">>> Plow capacity in %",  # display on hover
    50,  # default
    0,  # min
    100,  # max
)

gen.add(
    "debug_overlay",  # variable name
    bool_t,  # data type
    0,  # msg level
    ">>> Displays useful information for operator",  # display on hover
    False,  # default
)

gen.add(
    "e_stop",  # variable name
    bool_t,  # data type
    0,  # msg level
    ">>> E-STOP Stops all motion of the robot",  # display on hover
    False,  # default
)

# Generate the menu in rqt tool
exit(gen.generate(PACKAGE, "snow_blower", "SnowBlower"))
# 1st param - Package name constant from top
# 2nd param - node name this could be called in (for documentation only)
# 3rd param - This file's name without extension. Adds 'Config' at the end
#           - Use term_project_pkg.cfg and SnowBlowerConfig in py script

# for details: https://wiki.ros.org/dynamic_reconfigure/Tutorials/HowToWriteYourFirstCfgFile
